File: service-nodejs/apiAttendance/utils/tracer.js
const { initTracer } = require('jaeger-client')

//set up our tracer
const config = {
  serviceName: 'attendace-api',
  reporter: {
    logSpans: true,
    collectorEndpoint: process.env.JAEGER_COLLECTTOR_END_POINT,
  },
  sampler: {
    type: 'const',
    param: 1
  }
}
const options = {
  tags: {
    'attendace-api': '1.0.0'
  }
}
const tracer = initTracer(config, options);

module.exports = { 
  tracer
}

File: service-golang/utils/tracer.go
package utils

import (
	"io"
	"time"

	"github.com/getsentry/sentry-go"
	"github.com/opentracing/opentracing-go"
	"github.com/uber/jaeger-client-go/config"
)

func GetJaegerTracer(JaegerHostPort string, ServiceName string) (opentracing.Tracer, io.Closer, error) {

	cfg := config.Configuration{
		ServiceName: ServiceName,
		Sampler: &config.SamplerConfig{
			Type:  "const",
			Param: 1,
		},
		Reporter: &config.ReporterConfig{
			LogSpans:            false,
			BufferFlushInterval: 1 * time.Second,
			LocalAgentHostPort:  JaegerHostPort,
		},
	}
	tracer, closer, err := cfg.NewTracer()
	return tracer, closer, err
}

func SentryTracer(SentryDsn string) error {
	err := sentry.Init(sentry.ClientOptions{
		Dsn: SentryDsn,
	})
	return err
}
